src/lib/number.test.ts

```diff
@@ -0,0 +1,19 @@
+import { formatNumber } from './number'; // replace with your actual file name
+
+describe('formatNumber', () => {
+  it('should commify numbers less than 1 million', () => {
+    expect(formatNumber(12345)).toEqual('12,345');
+    expect(formatNumber('12345')).toEqual('12,345');
+  });
+
+  it('should format numbers less than 1 million with at most two decimal places', () => {
+    expect(formatNumber(12345.694206942)).toEqual('12,345.69');
+    expect(formatNumber('12345.6942069420')).toEqual('12,345.69');
+    expect(formatNumber('12345.6')).toEqual('12,345.6');
+  });
+
+  it('should format numbers greater than or equal to 1 million with suffix', () => {
+    expect(formatNumber(12345678)).toEqual('12.35M');
+    expect(formatNumber('12345678')).toEqual('12.35M');
+  });
```

Comment:
is this something we actually want? i didn't see the designs, but this feels inconsistent with how we're representing token amounts.

Reply:
The following was in the Jira issue:

> Truncate long fractional numbers – likely to arise one users can redeem rewards – and full integer values by rounding up and incorporating English words, i.e.,  (show 2 decimals) 12345.6942069420 → 12,345.70 or 12,345.69, 12345678901112 → 12,345,678,901,112 or 12.35 Trillion (bonus if simple)

I interpreted this as if we're wanting to millify large numbers, but could be wrong here!

Reply:
Just found [this Figma frame](https://www.figma.com/file/aETpyuG2AKjNcQtCjldcX0/ZERO-Messenger-%2F-Web?type=design&node-id=9703%3A268151&mode=design&t=iBCZXrRXxzczRa8n-1), which says we should millify 6 figures and above in the rewards popup

Reply:
🤷 well there you go.